Sources/MessagingInApp/Gist/Gist.swift

```diff
@@ -126,6 +139,41 @@ public class Gist: GistDelegate {
             }
     }
 
+    // If someone sets a page rule on a message, they want the message to show on that screen. Because messages can take multiple seconds to finish rendering, there is a chance that
+    // a user navigates away fron a screen when the rendering finishes. To fix this, cancel showing a modal message if a message is still loading.
+    //
+    // Like dismiss message, but does not call event listener.
+    // Dismiss the currently shown message, if there is one, and then remove message manager allowing us to show a message again in the future.
+    func cancelLoadingModalMessage() {
+        guard let messageManagerToCancel = getModalMessageManager() else {
+            return // no message being shown or loading.
+        }
+        let currentMessage = messageManagerToCancel.currentMessage
+
+        if messageManagerToCancel.isShowingMessage {
+            // The modal is already visible, don't cancel it.
+            // This can prevent an infinite loop scenario:
+            // * page rule changed and that triggers showing a Modal
+            // * Modal message is displayed on screen
+            // * Modal being displayed triggers an auto screenview tracking event. This triggers a SDK page route change
+            // * Request to cancel modal message
+            // * Back to the foreground screen that originally triggered showing a Modal message...repeat...
+            return
+        }
```

Comment:
This code gets removed in another PR in stack https://github.com/customerio/customerio-ios/pull/732

Reply:
This code originally existed to prevent an infinite loop but it does not handle test case explained [in this PR description](https://github.com/customerio/customerio-ios/pull/732).